tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -103,11 +103,16 @@ def a11y_select(self, match_string):
         self.accessibility.click(self.marionette.find_element(*_close_button_locator))
         self.wait_for_select_closed(*_close_button_locator)
 
-    def tap_element_from_system_app(self, element=None, add_statusbar_height=False):
-        # Workaround for bug 1109213, where tapping on the button inside the app itself
+    def tap_element_from_system_app(self, element=None, add_statusbar_height=False, x=None, y=None):        # Workaround for bug 1109213, where tapping on the button inside the app itself
         # makes Marionette spew out NoSuchWindowException errors
-        x = element.rect['x'] + element.rect['width']//2
-        y = element.rect['y'] + element.rect['height']//2
+        if x is None:
+            x = element.rect['x'] + element.rect['width']//2
+        else:
+            x = element.rect['x'] + x
+        if y is None:
+            y = element.rect['y'] + element.rect['height']//2
+        else:
+            y = element.rect['x'] + y
```

Comment:
Hrm, except this isn't working, because x and y are never None in this case, because we've just defined them with x = element.rect['x'] and y = element.rect['y']